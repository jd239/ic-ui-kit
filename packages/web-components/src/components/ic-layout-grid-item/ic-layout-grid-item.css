:host {
  grid-column: var(--ic-grid-item-col-start, 1) / span
    var(--ic-grid-item-col-span, var(--ic-layout-grid-col-span, 1)) !important;
  grid-row: var(--ic-grid-item-row-start, 1) / span
    var(--ic-grid-item-row-span, var(--ic-layout-grid-row-span, 1)) !important;
}

@media only screen and (max-width: 576px) {
  :host(.ic-layout-grid-hide-in-mobile) {
    display: none;
  }
}
