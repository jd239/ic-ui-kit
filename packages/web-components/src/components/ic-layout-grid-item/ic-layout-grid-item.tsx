import { Component, Prop, Host, h, Element, Watch } from "@stencil/core";

@Component({
  tag: "ic-layout-grid-item",
  styleUrl: "ic-layout-grid-item.css",
  shadow: true,
})
export class LayoutGridItem {
  @Element() el: HTMLIcLayoutGridItemElement;

  /**
   * The number of columns the grid item should span.
   */
  @Prop() colSpan?: number = 1;
  @Watch("colSpan")
  watchColSpan(newValue: number) {
    this.el.style.setProperty("--ic-grid-item-col-span", `${newValue}`);
  }

  /**
   * The column the grid item should start at.
   */
  @Prop() colStart?: number = 1;
  @Watch("colStart")
  watchColStart(newValue: number) {
    this.el.style.setProperty("--ic-grid-item-col-start", `${newValue}`);
  }

  /**
   * If `true`, the grid item will be hidden on smaller screens.
   */
  @Prop() hideInMobileMode?: boolean = false;

  /**
   * The number of rows the grid item should span.
   */
  @Prop() rowSpan?: number = 1;
  @Watch("rowSpan")
  watchRowSpan(newValue: number) {
    this.el.style.setProperty("--ic-grid-item-row-span", `${newValue}`);
  }

  /**
   * The row the grid item should start at.
   */
  @Prop() rowStart?: number = 1;
  @Watch("rowStart")
  watchRowStart(newValue: number) {
    this.el.style.setProperty("--ic-grid-item-row-start", `${newValue}`);
  }

  componentWillLoad(): void {
    this.el.style.setProperty("--ic-grid-item-col-start", `${this.colStart}`);
    this.el.style.setProperty("--ic-grid-item-col-span", `${this.colSpan}`);
    this.el.style.setProperty("--ic-grid-item-row-start", `${this.rowStart}`);
    this.el.style.setProperty("--ic-grid-item-row-span", `${this.rowSpan}`);
  }

  render() {
    const { hideInMobileMode } = this;
    return (
      <Host
        class={{
          "ic-layout-grid-hide-in-mobile": !!hideInMobileMode,
        }}
      >
        <slot></slot>
      </Host>
    );
  }
}
