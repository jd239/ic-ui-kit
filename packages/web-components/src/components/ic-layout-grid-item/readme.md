# ic-layout-grid-item



<!-- Auto Generated Below -->


## Properties

| Property           | Attribute             | Description                                                 | Type                   | Default |
| ------------------ | --------------------- | ----------------------------------------------------------- | ---------------------- | ------- |
| `colSpan`          | `col-span`            | The number of columns the grid item should span.            | `number \| undefined`  | `1`     |
| `colStart`         | `col-start`           | The column the grid item should start at.                   | `number \| undefined`  | `1`     |
| `hideInMobileMode` | `hide-in-mobile-mode` | If `true`, the grid item will be hidden on smaller screens. | `boolean \| undefined` | `false` |
| `rowSpan`          | `row-span`            | The number of rows the grid item should span.               | `number \| undefined`  | `1`     |
| `rowStart`         | `row-start`           | The row the grid item should start at.                      | `number \| undefined`  | `1`     |


----------------------------------------------


