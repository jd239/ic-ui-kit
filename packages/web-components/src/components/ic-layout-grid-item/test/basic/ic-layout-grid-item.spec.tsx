import { LayoutGridItem } from "../../ic-layout-grid-item";
import { newSpecPage } from "@stencil/core/testing";

describe("ic-layout-grid", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [LayoutGridItem],
      html: `<ic-layout-grid>
        <ic-layout-grid-item>
          <div>Test div 1</div>
        </ic-layout-grid-item>
      </ic-layout-grid>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with col start", async () => {
    const page = await newSpecPage({
      components: [LayoutGridItem],
      html: `<ic-layout-grid>
        <ic-layout-grid-item col-start="2">
          <div>Test div 1</div>
        </ic-layout-grid-item>
      </ic-layout-grid>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with row start", async () => {
    const page = await newSpecPage({
      components: [LayoutGridItem],
      html: `<ic-layout-grid>
        <ic-layout-grid-item row-start="2">
          <div>Test div 1</div>
        </ic-layout-grid-item>
      </ic-layout-grid>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with col span", async () => {
    const page = await newSpecPage({
      components: [LayoutGridItem],
      html: `<ic-layout-grid>
        <ic-layout-grid-item col-span="2">
          <div>Test div 1</div>
        </ic-layout-grid-item>
      </ic-layout-grid>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with row span", async () => {
    const page = await newSpecPage({
      components: [LayoutGridItem],
      html: `<ic-layout-grid>
        <ic-layout-grid-item row-span="2">
          <div>Test div 1</div>
        </ic-layout-grid-item>
      </ic-layout-grid>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with hide in mobile mode class", async () => {
    const page = await newSpecPage({
      components: [LayoutGridItem],
      html: `<ic-layout-grid>
        <ic-layout-grid-item hide-in-mobile-mode="true">
          <div>Test div 1</div>
        </ic-layout-grid-item>
      </ic-layout-grid>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should update --ic-grid-item-col-span when colSpan changes", async () => {
    const page = await newSpecPage({
      components: [LayoutGridItem],
      html: `<ic-layout-grid-item></ic-layout-grid-item>`,
    });
    page.rootInstance.colSpan = 5;
    await page.waitForChanges();
    expect(page.root!.style.getPropertyValue("--ic-grid-item-col-span")).toBe(
      "5"
    );
  });

  it("should update --ic-grid-item-col-start when colStart changes", async () => {
    const page = await newSpecPage({
      components: [LayoutGridItem],
      html: `<ic-layout-grid-item></ic-layout-grid-item>`,
    });
    page.rootInstance.colStart = 5;
    await page.waitForChanges();
    expect(page.root!.style.getPropertyValue("--ic-grid-item-col-start")).toBe(
      "5"
    );
  });

  it("should update --ic-grid-item-row-span when rowSpan changes", async () => {
    const page = await newSpecPage({
      components: [LayoutGridItem],
      html: `<ic-layout-grid-item></ic-layout-grid-item>`,
    });
    page.rootInstance.rowSpan = 5;
    await page.waitForChanges();
    expect(page.root!.style.getPropertyValue("--ic-grid-item-row-span")).toBe(
      "5"
    );
  });

  it("should update --ic-grid-item-row-start when rowStart changes", async () => {
    const page = await newSpecPage({
      components: [LayoutGridItem],
      html: `<ic-layout-grid-item></ic-layout-grid-item>`,
    });
    page.rootInstance.rowStart = 5;
    await page.waitForChanges();
    expect(page.root!.style.getPropertyValue("--ic-grid-item-row-start")).toBe(
      "5"
    );
  });
});
