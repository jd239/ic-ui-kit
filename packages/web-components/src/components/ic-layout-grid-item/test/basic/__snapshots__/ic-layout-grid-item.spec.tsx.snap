// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-layout-grid should render 1`] = `
<ic-layout-grid-item style="--ic-grid-item-col-start: 1; --ic-grid-item-col-span: 1; --ic-grid-item-row-start: 1; --ic-grid-item-row-span: 1;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <div>
    Test div 1
  </div>
</ic-layout-grid-item>
`;

exports[`ic-layout-grid should render with col span 1`] = `
<ic-layout-grid-item col-span="2" style="--ic-grid-item-col-start: 1; --ic-grid-item-col-span: 2; --ic-grid-item-row-start: 1; --ic-grid-item-row-span: 1;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <div>
    Test div 1
  </div>
</ic-layout-grid-item>
`;

exports[`ic-layout-grid should render with col start 1`] = `
<ic-layout-grid-item col-start="2" style="--ic-grid-item-col-start: 2; --ic-grid-item-col-span: 1; --ic-grid-item-row-start: 1; --ic-grid-item-row-span: 1;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <div>
    Test div 1
  </div>
</ic-layout-grid-item>
`;

exports[`ic-layout-grid should render with hide in mobile mode class 1`] = `
<ic-layout-grid-item class="ic-layout-grid-hide-in-mobile" hide-in-mobile-mode="true" style="--ic-grid-item-col-start: 1; --ic-grid-item-col-span: 1; --ic-grid-item-row-start: 1; --ic-grid-item-row-span: 1;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <div>
    Test div 1
  </div>
</ic-layout-grid-item>
`;

exports[`ic-layout-grid should render with row span 1`] = `
<ic-layout-grid-item row-span="2" style="--ic-grid-item-col-start: 1; --ic-grid-item-col-span: 1; --ic-grid-item-row-start: 1; --ic-grid-item-row-span: 2;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <div>
    Test div 1
  </div>
</ic-layout-grid-item>
`;

exports[`ic-layout-grid should render with row start 1`] = `
<ic-layout-grid-item row-start="2" style="--ic-grid-item-col-start: 1; --ic-grid-item-col-span: 1; --ic-grid-item-row-start: 2; --ic-grid-item-row-span: 1;">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <div>
    Test div 1
  </div>
</ic-layout-grid-item>
`;
